src/key/key.js

```diff
@@ -26,826 +26,836 @@
  */
 
 import armor from '../encoding/armor';
-import packet from '../packet';
+import {
+  PacketList,
+  PublicKeyPacket,
+  PublicSubkeyPacket,
+  SignaturePacket,
+  LiteralDataPacket
+} from '../packet';
 import enums from '../enums';
 import util from '../util';
 import User from './user';
 import SubKey from './subkey';
 import * as helper from './helper';
 
 /**
- * @class
- * @classdesc Class that represents an OpenPGP key. Must contain a primary key.
+ * Class that represents an OpenPGP key. Must contain a primary key.
  * Can contain additional subkeys, signatures, user ids, user attributes.
- * @param  {module:packet.List} packetlist The packets that form this key
- * @borrows module:packet.PublicKey#getKeyId as Key#getKeyId
- * @borrows module:packet.PublicKey#getFingerprint as Key#getFingerprint
- * @borrows module:packet.PublicKey#hasSameFingerprintAs as Key#hasSameFingerprintAs
- * @borrows module:packet.PublicKey#getAlgorithmInfo as Key#getAlgorithmInfo
- * @borrows module:packet.PublicKey#getCreationTime as Key#getCreationTime
- * @borrows module:packet.PublicKey#isDecrypted as Key#isDecrypted
- */
-export default function Key(packetlist) {
-  if (!(this instanceof Key)) {
-    return new Key(packetlist);
-  }
-  // same data as in packetlist but in structured form
-  this.keyPacket = null;
-  this.revocationSignatures = [];
-  this.directSignatures = [];
-  this.users = [];
-  this.subKeys = [];
-  this.packetlist2structure(packetlist);
-  if (!this.keyPacket || !this.users.length) {
-    throw new Error('Invalid key: need at least key and user ID packet');
+ * @borrows PublicKeyPacket#getKeyId as Key#getKeyId
+ * @borrows PublicKeyPacket#getFingerprint as Key#getFingerprint
+ * @borrows PublicKeyPacket#hasSameFingerprintAs as Key#hasSameFingerprintAs
+ * @borrows PublicKeyPacket#getAlgorithmInfo as Key#getAlgorithmInfo
+ * @borrows PublicKeyPacket#getCreationTime as Key#getCreationTime
+ * @borrows PublicKeyPacket#isDecrypted as Key#isDecrypted
+ */
+class Key {
```

Comment:
we could use `export default class Key` here.

Reply:
Yeah. bc6ddc6078464cf44d3b361af0c4c08ca6ee6a6d just mechanically changed prototype based classes to ES6 classes, and didn't change anything else. We can make things prettier later.